packages/explat-client/src/internal/experiment-assignments.ts

```diff
@@ -42,15 +27,17 @@ const fallbackExperimentAssignmentTtl = 60;
 
 /**
  * A fallback ExperimentAssignment we return when we can't retrieve one.
+ * As it is used in fallback situations, this function must never throw.
  *
  * @param experimentName The name of the experiment
  */
 export const createFallbackExperimentAssignment = (
 	experimentName: string
 ): ExperimentAssignment => ( {
-	experimentName,
+	experimentName: isName( experimentName ) ? experimentName : 'fallback_experiment_assignment',
```

Comment:
Under what circumstances would we call this with an invalid experiment name?

Reply:
Ah, I see it in #49832. I think it's problematic because it's a surprising result to get an assignment with a different name from what the calling code requested. 

By the way, this is the sort of change that I think would make more sense as part of #49832 (i.e., where it's needed), as trying to keep the context in mind across PRs over multiple weeks is pretty hard. It'd make it easier to review if we tried to minimise the number of open PRs by deploying often.

Reply:
> By the way, this is the sort of change that I think would make more sense as part of #49832 (i.e., where it's needed), as trying to keep the context in mind across PRs over multiple weeks is pretty hard. It'd make it easier to review if we tried to minimise the number of open PRs by deploying often.

This PR is almost entirely changes that are needed for #49832, split so that it won't be so big.

> Ah, I see it in #49832. I think it's problematic because it's a surprising result to get an assignment with a different name from what the calling code requested.

Yeah so this is necessary for experiments with invalid experiment names, we want to always have a valid ExperimentAssignment. I could add an `originalExperimentName` to keep the original name around.